src/App.tsx

```diff
@@ -15,47 +16,153 @@ export const goodsFromServer = [
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  if (sortType) {
+    visibleGoods.sort((a, b) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
+type Props = {};
+
+export const App: React.FC<Props> = () => {
+  // state = {
+  //   isReversed: false,
+  //   sortType: SortType.NONE,
+  // };
+
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  // handleAlpha = () => {
+  //   setSortType(
+  //     SortType.ALPHABET,
+  //   );
+  // };
+
+  // handleLength = () => {
+  //   this.setState({
+  //     sortType: SortType.LENGTH,
+  //   });
+  // };
+
+  // handleReverse = () => {
+  //   this.setState((state) => ({
+  //     isReversed: !state.isReversed,
+  //   }));
+  // };
+
+  const resetHandle = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
+  // render(): React.ReactNode {
+  //   const { isReversed, sortType } = this.state;
+
+  const visibleGoods
+    = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
+          onClick={() => setSortType(SortType.ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button is-success',
+            {
+              'is-light': sortType !== SortType.LENGTH,
+            },
+          )}
+          onClick={() => setSortType(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button is-warning',
+            {
+              'is-light': !isReversed,
+            },
+          )}
+          onClick={() => setIsReversed(current => !current)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortType !== SortType.NONE) && (
```

Comment:
I would recommend save this (isReversed || sortType !== SortType.NONE) into a variable befor the return() and use the variable here.

Reply:
agree with @kenvastolord